src/features/commands.ts

```diff
@@ -669,6 +669,56 @@ Instead:
       await tsk.delete();
     },
   },
+  {
+    words: ["!lock"],
+    help: "",
+    category: "Communication",
+    handleMessage: async (msg) => {
+      if (!msg.guild) {
+        return;
+      }
+
+      if (!isStaffAdmin(msg.member)) {
+        return;
+      }
+
+      // permission overwrites can only be applied on Guild Text Channels
+      if (msg.channel.type === "GUILD_TEXT") {
```

Comment:
Maybe:
```suggestion
      if (!msg.guild || !isStaffAdmin(msg.member)) {
        return;
      }

      // permission overwrites can only be applied on Guild Text Channels
      if (!msg.channel.isText()) {
```

Reply:
Oh, but if it needs to be a "guild text" maybe that doesn't work? Not sure if there's a similar check for DM/guild channel

Reply:
Yes,  it has to be narrowed to `guild_text` because `isText` implies it can also be `DM | Thread` and you cannot have permission overwrites on it